core/services/chainlink/config_general_secrets.go

```diff
@@ -32,3 +38,29 @@ func (g *generalConfig) PyroscopeAuthToken() string {
 	}
 	return string(*g.secrets.Pyroscope.AuthToken)
 }
+
+func (g *generalConfig) MercuryCredentials(url string) (username, password string, err error) {
+	if g.secrets.Mercury.Credentials == nil {
+		return "", "", errors.New("no Mercury credentials were specified in the config")
+	}
+	for _, creds := range g.secrets.Mercury.Credentials {
+		if creds.URL != nil && creds.URL.URL().String() == url {
+			if creds.Username == nil {
+				return "", "", errors.Errorf("no Mercury username specified for server URL: %q", url)
+			}
+			if creds.Password == nil {
+				return "", "", errors.Errorf("no Mercury password specified for server URL: %q", url)
+			}
+			return string(*creds.Username), string(*creds.Password), nil
```

Comment:
so it just picks the first it finds. is this due to the nature on how TOML parsing works?

Reply:
It will pick the first one that matches the URL. We probably need unique enforcement for this 🤔